airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/stream.py

```diff
@@ -156,11 +154,10 @@ def get_time_ordered_file_infos(self) -> List[FileInfo]:
         return sorted(self.pattern_matched_filepath_iterator(self.filepath_iterator()), key=lambda file_info: file_info.last_modified)
 
     def _get_schema_map(self) -> Mapping[str, Any]:
-        return_schema: Dict[str, Any] = None
-        if self._schema != {}:
+        if self._schema:
             return_schema = deepcopy(self._schema)
         else:  # we have no provided schema or schema state from a previous incremental run
-            return_schema = self._get_master_schema()
+            return_schema = self.master_schema if self.master_schema else self._get_master_schema()
```

Comment:
This change is unnecessary since this check already happens in `self._get_master_schema()`